url4c41254592_topskychen_AuthTopk_tgz-pJ8-quickhull3d_QuickHull3DTestJ8/src/utility/geo/genDataOfLineForHadoop.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * 
+ */
+package utility.geo;
+
+/**
+ * @author qchen
+ *
+ */
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Scanner;
+import java.util.Set;
+
+import jdbm.PrimaryTreeMap;
+import jdbm.RecordManager;
+import jdbm.RecordManagerFactory;
+import utility.Compare.DataOfPoint;
+import utility.Compare.DataOfPointParser;
+import utility.Compare.buildBtreeOfPoints;
+
+/**
+ * remember first build buildBtreeOfPoints to get delaynay information
+ * @author qchen
+ * */
+public class genDataOfLineForHadoop {
+	public static boolean DEBUG = true;
+	public static String filename = "database/LinesData.NE.5HOP.tmp";
+	
+	public void loadData(int k) throws IOException{
+		RecordManager recmanOfPoints = RecordManagerFactory.createRecordManager(DataOfPointParser.filename);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *recmanOfPoints*. The resource is referenced at line 41. The resource is closed at line 81. There are other execution paths that do not contain closure statements, for example, when DataOfPoint.loadFromBytes throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).